.circleci/config.yml

```diff
@@ -1,158 +1,165 @@
 version: 2.1
 orbs:
   aws-s3: circleci/aws-s3@2.0.0
-
-commands:
-  source-vars:
-    description: source the environment variables from the temp workspace
-    steps:
-      - run: |
-          set -x
-          cat /tmp/workspace/env_vars.sh >> $BASH_ENV
-          source $BASH_ENV
+parameters:
+  cross-container-tag:
+    type: string
+    default: go1.15.10-ec819fab057e2efdcf478fec05bb5533957721ec
 
 jobs:
-  export_vars:
-    machine:
-      enabled: true
-      docker_layer_caching: true
-    steps:
-      - checkout
-      - run:
-          name: Export environment variables to bash environment
-          command: |
-            set -x
-            go_ver=$(grep -r '^go ' go.mod | cut -d ' ' -f2)
-            test -n "$go_ver"
-            working_dir=/home/circleci/project
-            touch env_vars.sh
-            echo "export WORKING_DIR=$working_dir" >> env_vars.sh
-            echo "export GO_VERSION=$go_ver" >> env_vars.sh
-      - persist_to_workspace:
-          root: .
-          paths:
-            - env_vars.sh
   build_binaries:
-    machine:
-      enabled: true
-      docker_layer_caching: true
+    docker:
+      - image: quay.io/influxdb/cross-builder:<< pipeline.parameters.cross-container-tag >>
     steps:
       - attach_workspace:
           at: /tmp/workspace
-      - source-vars
       - checkout
+      - restore_cache:
+          keys:
+            - influxdb-cache-v1-{{ checksum "go.mod" }}
+            - influxdb-cache-v1
       - run:
-          name: Build source tarballs
+          name: Build source tarball
           command: |
             set -x
-            mkdir -p ${WORKING_DIR}/tarball
-            "${WORKING_DIR}/releng/source-tarball/build.bash" \
-              -p "${WORKING_DIR}/.git" \
-              -s "${CIRCLE_SHA1}" \
-              -b "${CIRCLE_BRANCH}" \
-              -v "${CIRCLE_BRANCH}-${CIRCLE_SHA1}" \
-              -o "${WORKING_DIR}/tarball"
+            mkdir -p ./tarball
+            TARBALL_NAME="./tarball/influxdb-src-${CIRCLE_SHA1}.tar.gz"
+            tar --exclude-vcs --exclude './tarball' -vczf "${TARBALL_NAME}" .
       - store_artifacts:
           path: tarball/
       - run:
           name: Build binaries
           command: |
-            OUTDIR=${WORKING_DIR}/bins
-            mkdir -p ${OUTDIR}
-            GOOS=linux GOARCH=amd64 "${WORKING_DIR}/releng/raw-binaries/build.bash" \
-              -i "${WORKING_DIR}/tarball/influxdb-src-${CIRCLE_SHA1}.tar.gz" \
-              -o "$OUTDIR"
+            set -x
+
+            tarsum()
+            {
+              FROM_DIR=$1
+              TARBALL_PATH=$2
+              tar -C $FROM_DIR -cvzf ${TARBALL_PATH} .
+              md5sum ${TARBALL_PATH} > ${TARBALL_PATH}.md5
+              sha256sum ${TARBALL_PATH} > ${TARBALL_PATH}.sha256
+            }
+
+            # TODO: chmod should be done in top-level container
```

Comment:
What are the consequences of leaving this as a TODO in this PR?

Reply:
None - but fixed in https://github.com/influxdata/influxdb/pull/22333